azure-iot-device/azure/iot/device/iothub/auth/base_renewable_token_authentication_provider.py

```diff
@@ -161,6 +161,7 @@ def timerfunc():
             self.generate_new_sas_token()
 
         self._token_update_timer = Timer(seconds_until_update, timerfunc)
+        self._token_update_timer.daemon = True
```

Comment:
>self._token_update_timer.daemon = True [](start = 8, length = 38)

as a side-effect, we can remove the disconnect function from this class.  The only reason for disconnect to exist was to kill the timer, but now we don't have to worry about this.